### PR TITLE
Makes undeads and androids have no metabolism

### PR DESCRIPTION
**Pull request by XDTM**
## About The Pull Request

Added two new traits, TRAIT_NOMETABOLISM and TRAIT_TOXIMMUNE. Does what it says on the tin, making livers not process reagents at all (except liverless reagents) and gives immunity to toxin damage. Species with NOMETABOLISM spawn with no liver (NOLIVER is now redundant and has been removed). This trait also prevents liver failure damage, for obvious reasons.

These traits have been given by default to androids (artificial, they were already immune to chemical healing anyway), zombies (who have their own regeneration) and skeletons (milk still works).

Other changes:
- Species' handle_reagents proc now fires before checking for metabolization, so species can process chems even when liverless.
- Removed the calcium healer trait, making it into a species handle_reagents check for each affected species (skeletons, plasmamen, bone golems). Skeletons now also heal burn damage from milk, since it's one of the few forms of healing they have available.

_**Note:**_
These traits should logically be applied to plasmamen and golems, but i intentionally left them out for now since it would be a significant balance shift, and there should be a proper alternative way of healing them first.

## Why It's Good For The Game

Balances the significant benefits of being undead with a significant negative: being immune to chemical healing. Since zombies don't really rely on it and androids were already heal-immune, this mostly affects liches, and i believe it should be fine: considering that the main point of the spell is the resurrection mechanism, the space, gas, pierce, and heat/cold immunity they also gain should be counterbalanced by having a harder time recovering from injuries.

There is also a silver lining to having this trait: poison immunity. Effectively this is a buff to androids and zombies, the latter especially because they can no longer process mutation toxins. While this tactic is creative, it also trivializes a very expensive investment of TC. I might still add a liverless way to apply mutation toxins later on, but zombies shouldn't be effectively one-shot by a syringe gun.

## Changelog
:cl:
add: Androids, skeletons and zombies no longer metabolize reagents. As such they no longer benefit from healing reagents, nor are affected by poisons and toxins.
add: These species are now also immune to any other form of toxin damage.
tweak: Milk now also heals burn damage for skeletons.
/:cl:

### CODE REVIEW

**Comment by Tlaltecuhtli** [comment]
is the krokodyl zombie affected ?

**Comment by XDTM** [comment]
nope

**Comment by CPTANT** [comment]
I am not really sure how a skeleton would heal toxins damage now.

**Comment by Tlaltecuhtli** [comment]
why do skeletons have toxin damage? they already turn oxy into brute or something like that

**Comment by CPTANT** [comment]
Radiation? Zombie tumors? virusses?

I have honestly no idea what works and doesn't work on them but its very easy for a source of toxin damage to slip in. Making them convert/heal all toxins damage automatically would make the code robust against oversights.

**Comment by XDTM** [comment]
Good point, i'm considering a notoxins trait to pair with the nometabolism. I'm not making in part of nometabolism itself, since that trait could be gained in the future as a temporary effect by liver-owning beings.

Unrelated fun fact: these species can metabolize normally if aided by a metabolism virus!

**Comment by knacker48** [comment]
Wait will this make so you can't use mutation toxins to cure zombies?

**Comment by XDTM** [comment]
Yeah, their effect requires metabolization.

**Comment by Angustmeta** [comment]
I see you added exceptions for bone hurting juice, but will the OD effects still work?

**Comment by knacker48** [comment]
Hmm mutation toxins were my main way of dealing with zombies before. Would you be willing to make an exception for them (like how you are with bone hurting juice) or do you think that zombies need the buff?

**Comment by ExcessiveUseOfCobblestone** [comment]
do the intended surgery rather than cheesing it.

**Comment by FantasticFwoosh** [comment]
Big wizard nerf now, they need those medpacks or the lich is going to spend more time being dead and vunerable to having their phylactery discovered.

**Comment by fludd12** [comment]
I picture this prompting _interesting_ reactions from lich-wizards who manage to get a chemistry setup, as mass producing black powder pills is not especially difficult.

**Comment by XDTM** [comment]
Can't wait to hear the tales of terrorist wizards paying back the crew for all the suicide bombings they endured in the past years

**Comment by ATH1909** [comment]
> I am not really sure how a skeleton would heal toxins damage now.

Could we make milk also heal tox damage? IIRC, ebows and such still directly deal tox damage without involving chems.

Does the regen of zombies currently heal tox damage? If not, we might want to let zombies naturally regenerate tox damage.

What about clone damage? Does rezadone work on things that don't have a metabolism?

**Comment by necromanceranne** [comment]
Fwoosh how the fuck is this a wizard NERF. This buffs wizards if anything by making liches be immune to the most surefire way to kill a wizard. Shoot them with lethal chems.

**Comment by XDTM** [comment]
they were already pierceimmune, so it doesn't change all that much

**Comment by necromanceranne** [comment]
Doesn't piercing syringes bypass that?

**Comment by XDTM** [comment]
I suppose, but I'm not sure if research can consistently reach those during a wizard round. 
In any case chems are the weapons of choice against wizards because damage usually doesn't stick, since they can raid medbay and teleport away to heal when harmed. If they cannot heal, conventional weapons will eventually take them down.

**Comment by ArcaneMusic** [comment]
>
> In any case chems are the weapons of choice against wizards because damage usually doesn't stick, since they can raid medbay and teleport away to heal when harmed. If they cannot heal, conventional weapons will eventually take them down.

For posterity's sake, the method of choice for power-gaming wizards is still suicidally welder-bombing the wizard and then having the rest of the crew beat the wizard to death.
Off that note, good change. Dead things should act dead.

File: code/modules/mob/living/carbon/damage_procs.dm [hunk 1]
```diff
@@ -87,6 +87,8 @@
 			blood_volume -= 5*amount
 		else
 			blood_volume -= amount
+	if(HAS_TRAIT(src, TRAIT_TOXIMMUNE)) //Prevents toxin damage, but not healing
```
Comment (optimumtact) [inline code/modules/mob/living/carbon/damage_procs.dm]: is this an undocumented change?

Comment (XDTM) [reply code/modules/mob/living/carbon/damage_procs.dm]: added later as a response to comment concerns, i forgot to add it to the PR desc

**Comment by XDTM** [comment]
@ExcessiveUseOfCobblestone fixed conflicts

**Pull request merged**